client/src/reducers/ui/layout.js

```diff
@@ -1,21 +1,33 @@
 import Immutable from 'seamless-immutable'
+import _ from 'lodash'
+
 import {
-  TOGGLE_LEFT,
+  SET_LEFT_OPEN_CLOSE,
   TOGGLE_RIGHT,
   TOGGLE_MAP,
 } from '../../actions/LayoutActions'
 
+import {LOCATION_CHANGE} from 'react-router-redux'
+
 export const initialState = Immutable({
   showLeft: true,
+  leftOpen: true,
   showRight: false,
   showMap: false,
 })
 
 export const layout = (state = initialState, action) => {
   switch (action.type) {
-    case TOGGLE_LEFT:
-      const previousShowLeft = state.showLeft
-      return Immutable.set(state, 'showLeft', !previousShowLeft)
+    case LOCATION_CHANGE:
+      const path = action.payload.pathname
+      const is508 =
```

Comment:
Since this logic is in at least two places, should it be a utility function somewhere?

Reply:
I completely agree :)
Take a look at my branch for 445 (which is the WIP for the granule list stuff)

Reply:
I cherry picked these from the other branch.